libs/core/src/lib/resizable-card-layout/resizable-card-layout.component.ts

```diff
@@ -0,0 +1,497 @@
+    OnInit,
+    Output,
+    QueryList,
+    ViewChild,
+    ViewEncapsulation
+} from '@angular/core';
+import { FocusKeyManager } from '@angular/cdk/a11y';
+import { takeUntil } from 'rxjs/operators';
+import { Subject } from 'rxjs';
+
+import {
+    CardDropped,
+    HorizontalResizeStep,
+    ResizingEvent,
+    ResizedEvent,
+    ResizableCardItemComponent,
+    ResizableCardItemConfig
+} from './resizable-card-item/resizable-card-item.component';
+
+const DRAG_START_DELAY = 500;
+export type LayoutSize = 'sm' | 'md' | 'lg' | 'xl';
+export type ResizableCardLayoutConfig = Array<ResizableCardItemConfig>;
+
+@Component({
+    selector: 'fd-resizable-card-layout',
+    templateUrl: 'resizable-card-layout.component.html',
+    styleUrls: ['./resizable-card-layout.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    encapsulation: ViewEncapsulation.None
+})
+export class ResizableCardLayoutComponent implements OnInit, AfterViewInit, AfterContentInit, OnDestroy {
+    /** Drag of card can be disabled. Default is true */
+    @Input()
+    draggable = true;
+
+    /** Object for setting each card property */
+    @Input()
+    layoutConfig: ResizableCardLayoutConfig;
+
+    /** Available layout width */
+    @Input()
+    get layoutSize(): LayoutSize {
+        return this._layout;
+    }
+
+    set layoutSize(layoutSize: LayoutSize) {
+        this._layout = layoutSize;
+        this._setLayoutColumns(layoutSize);
+    }
+
+    /** Emits when card resize is reached to one new step in horizontal or vertical direction */
+    @Output()
+    stepChange: EventEmitter<ResizedEvent> = new EventEmitter<ResizedEvent>();
+
+    /** Emits when card is dragged and dropped at different position in layout */
+    @Output()
+    dropped: EventEmitter<CardDropped> = new EventEmitter<CardDropped>();
+
+    /** Emits when card is still resizing */
+    @Output()
+    resizing: EventEmitter<ResizingEvent> = new EventEmitter<ResizingEvent>();
+
+    /** Emits when card resize is completed */
+    @Output()
+    resized: EventEmitter<ResizedEvent> = new EventEmitter<ResizedEvent>();
+
+    /** Emits when card height is reduced to show only header */
+    @Output()
+    miniHeaderReached: EventEmitter<ResizedEvent> = new EventEmitter<ResizedEvent>();
+
+    /** Emits when minimum height of card content area is reached */
+    @Output()
+    miniContentReached: EventEmitter<ResizedEvent> = new EventEmitter<ResizedEvent>();
+
+    /** Emits when layout changes */
+    @Output()
+    layoutChange: EventEmitter<ResizableCardLayoutConfig> = new EventEmitter<ResizableCardLayoutConfig>();
+
+    /** @hidden */
+    // @ViewChildren(ResizableCardItemComponent)
+    // resizeCardItems: QueryList<ResizableCardItemComponent>;
+
+    /** @hidden */
+    @ContentChildren(ResizableCardItemComponent)
+    resizeCardItems: QueryList<ResizableCardItemComponent>;
+
+    /** @hidden */
+    layoutWidth: number;
+
+    /** @hidden */
+    layoutHeight: number;
```

Comment:
all hidden props should be prefixed with `_`

Reply:
fixed